Samples~/LootLockerExamples/8 - Virtual Currency Purchases/LootLockerVirtualStore.cs

```diff
@@ -0,0 +1,126 @@
+using UnityEngine;
+using UnityEngine.UI;
+using LootLocker.Requests;
+
+namespace LootLocker
+{
+    public class LootLockerVirtualStore : MonoBehaviour
+    {
+        public static LootLockerVirtualStore Instance { get; private set; }
+
+        public Transform storeUI;
+        public Transform virtualPurchaseProductPrefab;
+
+        public Text goldAmount;
+
+        public string catalogKey = "";
+        public string walletID = "";
+
+        private string currencyCode;
+
+        public void Awake()
+        {
+
+            if (Instance == null)
+            {
+                Instance = this;
+            }
+            else Destroy(gameObject);
+
+            LootLockerSettingsOverrider.OverrideSettings();
+
+            LootLockerSDKManager.StartGuestSession((response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not start guest session");
+                    return;
+                }
+                FetchPlayerWallet(response.player_ulid);
+                FetchVirtualProductsFromCatalog();
+
+            });
+        }
+
+        public void FetchPlayerWallet(string ulid)
+        {
+            LootLockerSDKManager.GetWalletByHolderId(ulid, LootLockerEnums.LootLockerWalletHolderTypes.player, (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not start guest session");
+                    return;
+                }
+
+                walletID = response.id;
+                GetPlayerGold();
+                GiveGold();
+            });
+        }
+
+        public void GetPlayerGold()
+        {
+            LootLockerSDKManager.ListBalancesInWallet(walletID, (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not list balances");
+                    return;
+                }
+
+                currencyCode = response.balances[0].currency.code.ToUpper() + ": ";
+
+                goldAmount.text = currencyCode + response.balances[0].amount;
+            });
+        }
+
+
+        public void FetchVirtualProductsFromCatalog()
+        {
+            LootLockerSDKManager.ListCatalogItems(catalogKey, 0, "", (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not fetch Catalog Items");
+                    return;
+                }
+
+
+                foreach (var item in response.entries)
+                {
+
+                    switch(item.entity_kind){
+                        case LootLockerEnums.LootLockerCatalogEntryEntityKind.asset:
+                            var something = response.asset_details[item.GetItemDetailsKey()];
+                            Debug.Log(something.name);
+                            break;
+
+                    }
```

Comment:
Seems like debugging code?

Reply:
Yes you're right! I've removed it in the upcoming commit ty